### DISCUSSION TITLE
(14) Città Verde [P.20220701]

### DISCUSSION THREAD

**Discussion by progettistamatto** [sapienzastudentsnetwork/basi-di-dati-2#27] [Progetti]
In questa discussion a tema il progetto [Città Verde](https://elearning.uniroma1.it/pluginfile.php/1350504/mod_folder/content/0/%285%29%20P%20-%20Progetti/%2814%29%20Citt%C3%A0%20Verde.pdf), discuteremo di eventuali proposte di soluzione, dubbi o scelte progettuali in merito alla realizzazione del progetto in questione

**N.B.:** rispetto al testo oggetto dell'esame "<ins>GreenCare</ins>" del 3 settembre 2021 le specifiche del progetto P.20220701 ("Città Verde") differiscono per l'aggiunta della seguente specifica:

-  > È importante che il sistema tenga traccia delle malattie che colpiscono le piante della città. In particolare, di ogni malattia interessa conoscere il nome scientifico, quello “volgare” e la gravità (_bassa, media, alta, mortale_).

   > Di ogni pianta interessa quindi conoscere anche lo storico delle malattie che l’hanno colpita: di ogni episodio di malattia interessa la data di scoperta del problema e l’eventuale data di risoluzione; se il problema è risolto, interessa conoscere anche l’intervento in cui è stata somministrata la cura o si è verificata la guarigione.

E di un annesso ulteriore use case:

3. > Dato un periodo di tempo e un insieme di malattie, il sistema deve calcolare un resoconto che associ ad ognuna delle malattie indicate il suo tasso _ξ_. Il tasso _ξ_ di una malattia è il rapporto tra il numero di soggetti verdi che ha avuto tale malattia nel periodo dato e il numero di soggetti verdi che ha avuto una qualunque malattia nel periodo dato.

**Comment by progettistamatto** [comment]
Per questo progetto è disponibile [[QUI]](https://github.com/Exyss/university-notes/raw/main/Bachelor/Secondo%20Anno/Basi%20di%20Dati%20II/Citt%C3%A0%20Verde.pdf) una soluzione proposta da Simone Bianco (Exyss)

> [!Warning]
_Le soluzioni di Simone Bianco fanno uso degli **Entity-Relationship (ER)** Diagrams su cui il corso si basava fino all'Anno Accademico 2023/2024, mentre **attualmente vengono utilizzati gli UML Class** Diagram. Sebbene i due siano estremamente simili, vi sono **alcune discordanze tra di loro**, in particolare l'**inversione di posizione dei vincoli di molteplicità agli estremi delle relationship**. Nonostante ciò, il rimanente delle fasi di analisi e progettazione delle soluzioni proposte risulta inaffetto da tale cambiamento. Per tanto, una volta comprese le differenze tra ER e Classi UML, le soluzioni risultano normalmente fruibili._

Rispondi a questo commento taggando @Exyss per eventuali dubbi o segnalazioni in merito

**Comment by 0ri0nRo** [comment]
Stavo leggendo la soluzione proporsta da Simone Bianco ( @Exyss ) ma nello Use-Case che dato un Intervento trova le squadre disponibili non bisogna prendere anche le date? Magari una squadra è giusta ma è impegnata con un altro intervento.

**Reply by Exyss** [reply]
Tale caso è già gestito dal controllo "Esiste un sevizio che non è terminato"

![image](https://github.com/sapienzastudentsnetwork/bdii2223/assets/65043207/7c160803-6b9c-4687-b856-20030076c3a3)

**Reply by 0ri0nRo** [reply]
Ragionandoci non ha senso mettere nessun vincolo di data. Grazie mille!

**Comment by progettistamatto** [comment]
Per questo progetto è disponibile [[QUI]](https://github.com/SimoneLid/BD2/raw/main/Progetti/Citt%C3%A0%20Verde/Citt%C3%A0%20Verde.pdf) una soluzione proposta da Simone Lidonnici (@SimoneLid)

**Comment by CosaFaSelect1** [comment]
Per questo progetto è disponibile [[QUI]](https://github.com/CasuFrost/university-notes-Informatica/raw/main/Secondo%20Anno/Secondo%20Semestre/Basi%20di%20Dati%202/Progetti/Citt%C3%A0%20Verde/CittaVerde.pdf) una soluzione proposta da Marco Casu
